controllers/web.js

```diff
@@ -0,0 +1,193 @@
+const Bluebird = require("bluebird");
+//const responsesCst = require("../constants/responses").FR;
+const { ButtonsListMessage, Button} = require("../platforms/generics");
+const { camelCase } = require("lodash");
+const uuid = require("uuid/v4");
+
+module.exports = () => {
+
+  function postbackReceived(res, senderId, message) {
+    let needFeedback = false;
+    return bot.ask("postback", senderId, message, null, null, res)
+    .then( (answer)=> {
+      needFeedback = answer.feedback || needFeedback;
+      return sendResponse(res, senderId, answer.responses);
+    }).then( () => {
+      if (needFeedback) {
+        return sendFeedback(res, senderId, message, "message");
+      }
+    })
+    .catch( (err) => {
+      res.logger.error(err);
+      return sendResponse(res, "Oups : " + err.message);
+    });
+
+  }
+
+
+  function messageReceived(res, senderId, message) {
+    let needFeedback = false;
+
+    apiai.textRequestAsync(message, { sessionId: senderId })
+    .then( (resp) => {
+      let senderId = resp.sessionId;
+      if (resp.status && resp.status.code === 200 && resp.result) {
+        //successful request
+
+        //if user need to login
+        if (resp.result.action === "connection" || resp.result.action === "welcome") {
+          return sendResponse(res, senderId, new Button("web_url", `${config.server.url}${config.server.basePath}/authorize?state=${senderId}-web`,"Il faut d'abord se connecter !"));
+        }
+
+        //if api.ai has a premade response
+        if (resp.result.fulfillment && resp.result.fulfillment.speech && Array.isArray(resp.result.fulfillment.messages) && resp.result.fulfillment.messages.length) {
+          let smalltalk = resp.result.action && resp.result.action.indexOf("smalltalk") !== -1;
+          let quickResponses = resp.result.fulfillment.messages;
+
+          if (smalltalk && Math.floor((Math.random() * 2))) { //random to change response from original smalltalk to our custom sentence
+            quickResponses = [{ speech: resp.result.fulfillment.speech, type: 0 }];
+          }
+
+          return sendQuickResponses(res, senderId, quickResponses);
+        }
+        //else bot take over
+        return bot.ask("message", senderId, message, resp.result.action, resp.result.parameters, res)
+        .then( (answer) => {
+          needFeedback = answer.feedback || needFeedback;
+          return sendResponse(res, senderId, answer.responses).then( () => {
+            if (needFeedback) {
+              return sendFeedback(res, senderId, resp.result.action, message);
+            }
+          });
+        })
+        .catch( (err) => {
+          res.logger.error(err);
+          return sendResponse(res, "Oups : " + err.message);
+        });
+
+      }
+    });
+
+
+  }
+
+
+  function getHistory(req, res) {
+    let senderId = req.params.senderId;
+    if (!senderId) {
+      return res.status(200).end();
+    } else {
+      return web.getHistory(res, senderId);
```

Comment:
A better way is to return a promise into your web.getHistory and call res.status().json here